File: 0226-invert-binary-tree/0226-invert-binary-tree.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

function invertTree(root: TreeNode | null): TreeNode | null {
    if (!root) return root
    const stack: TreeNode[] = [root]
    while (stack.length) {
        const cur = stack.pop()
        const left = cur.left
        cur.left = cur.right
        cur.right = left
        if (cur.left) stack.push(cur.left)
        if (cur.right) stack.push(cur.right)
    }
    return root
    
    if (!root) return root
    const left = root.left
    root.left = invertTree(root.right)
    root.right = invertTree(left)
    return root
};

File: 0226-invert-binary-tree/0226-invert-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        if not root:
            return root
        left = root.left
        root.left = root.right
        root.right = left
        self.invertTree(root.left)
        self.invertTree(root.right)
        return root
        
        if not root:
            return root
        q = deque([root])
        while q:
            cur = q.popleft()
            left = cur.left
            cur.left = cur.right
            cur.right = left
            if cur.left:
                q.append(cur.left)
            if cur.right:
                q.append(cur.right)
        return root
        
        if not root:
            return root
        left = root.left
        root.left = self.invertTree(root.right)
        root.right = self.invertTree(left)
        return root